Proxy works with reviews service

Nick-Zabalza opened the pull request:
@margotTenenbaum  @GrumpyCat0709 

Hi Kate and Elizabeth,

I know the ReadMe says to get all the services running before starting my proxy server but that is anticipation of our services being complete. As of now, I have only tested my proxy with the Reviews service. Also, the proxy's HTML file is only coded to work with my Reviews component and so, even if you fire up other services, they won't mount. So you only need to use the Reviews component to test my proxy out.

A note on testing: My tests are bare minimum for the proxy since they rely on each service fully testing their components. All my tests will do, for each service, is search for an element with a unique ID and see if it exists. Then they search for another element with a unique ID that should have innerhTML unique to item 100. If those tests pass, then the proxy correctly mounted each service and each service correctly retrieved data. As of now, only those tests for the Reviews service are written.

Another note on testing: The testing method I used for the SpecRunner.html is probably something you haven't seen before. In that html file, there is an iframe that loads the proxy's index.html file fully. Which means it needs time to load the entire website before running the tests. In the tests, you will note a variable called "iframeDOM". The contents of an iframe aren't located in the normal DOM of a webpage. Instead, they reside within the iframe's own DOM structure. As the variable suggests, that DOM is saved to it. Yet you might wonder where that variable is being defined since it isn't in the tests.js. Definition is in the SpecRunner.html file.

margotTenenbaum commented:
Hi @Nick-Zabalza,

This looks good! I checked it with the reviews component running and everything worked as expected, with all tests passing. Very easy-to-follow instructions in the README. Just a few notes below:

-In index.js, is the cookieParser still needed? I commented it out to test and everything still seemed to work fine. If it is no longer needed you may want to remove it from your dependencies, etc.
-There is a typo on line 28 of README (visit) and 2 on line 30 (html) (the)
-This is more of a note for the future: I noticed you currently only have happy path tests for your proxy. If possible, you may want to consider adding a few unhappy paths as well.

Great work here, and once you are comfortable with the above points you may merge!

Nick-Zabalza commented:
@margotTenenbaum 

Thourough as usual, Elizabeth. Thank you.

I fixed typos, removed any mention of cookie parser, and put a comment in tests.js to remind me to write bad path tests.
